### DISCUSSION TITLE
graphql/codegen/docs/advanced/how-does-it-work

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#9118] [Docs Discussions]
# graphql/codegen/docs/advanced/how-does-it-work

GraphQL Code Generator Documentation

https://the-guild.dev/graphql/codegen/docs/advanced/how-does-it-work

**Comment by NgoranA** [comment]
Please i have this issue. 
I set up codegen.ts as requested along with ["typescript", "typescript-operations", "typescript-urql"] as plugins
In my graphql folder i have 
- user.operations.graphql
- user.schema.graphql
when i execute pnpm generate, i have duplicate types. what could be the reason?
Thank you.

**Reply by n1ru4l** [reply]
Hey, can you please share a full reproduction of the issue that you are facing? Normally types should only be generated once and it might seem that you are applying a plugin twice.

**Comment by zargold** [comment]
I presently use the graphql introspection query to get the available queries and types and use apollo codegen to generate types based only on the results of that introspection fetch request data. It seems that this type generator relies on using the original .schema files. In my case those are not available--would I still be able to generate types with this product?

**Reply by Urigo** [reply]
@zargold Codegen support getting the schema from all possible sources, including from introspection
You can find all options [here](https://the-guild.dev/graphql/codegen/docs/config-reference/schema-field#available-formats)

**Comment by sanalpanicker** [comment]
Hi Team, I am having an issue after I install graphiql package.
after I install graphiql package I cannot run yarn codegen anymore. it thorw's a syntax error :
⚠ Generate outputs
  ❯ Generate to ./src/types/graphql.ts
    ✔ Load GraphQL schemas
    ✖ Syntax Error: Expected Name, found "}".
    ◼ Generate

To test, just 'yarn add graphiql' to your existing project. try running yarn codegen - it will fail.
My codesandbox link here : https://codesandbox.io/p/sandbox/fragrant-bird-6n72vc

I also created an issue here : https://github.com/dotansimha/graphql-code-generator/issues/9755

**Comment by sanalpanicker** [comment]
how can I write conditional queries?
